builds/Makefile

```diff
@@ -618,96 +550,116 @@ IOS_BCL_TARGETS +=
+$(PREFIX)/lib/mono/2.1/%: $(BUILD_DESTDIR)/ios-bcl/monotouch/% | $(PREFIX)/lib/mono/2.1 $(PREFIX)/lib/mono/2.1/Facades
 	$(Q) install -m 0755 $< $@
 
-$(PREFIX)/lib/mono/Xamarin.WatchOS/repl/%: $(BUILD_DESTDIR)/bcl/monotouch_watch_runtime/% | $(PREFIX)/lib/mono/Xamarin.WatchOS/repl
+$(PREFIX)/lib/mono/Xamarin.WatchOS/repl/%: $(BUILD_DESTDIR)/ios-bcl/monotouch_watch_runtime/% | $(PREFIX)/lib/mono/Xamarin.WatchOS/repl
 	$(Q) install -m 0755 $< $@
 
-$(PREFIX)/lib/mono/Xamarin.WatchOS/%: $(BUILD_DESTDIR)/bcl/monotouch_watch/% | $(PREFIX)/lib/mono/Xamarin.WatchOS $(PREFIX)/lib/mono/Xamarin.WatchOS/Facades
+$(PREFIX)/lib/mono/Xamarin.WatchOS/%: $(BUILD_DESTDIR)/ios-bcl/monotouch_watch/% | $(PREFIX)/lib/mono/Xamarin.WatchOS $(PREFIX)/lib/mono/Xamarin.WatchOS/Facades
 	$(Q) install -m 0755 $< $@
 
-$(PREFIX)/lib/mono/Xamarin.TVOS/repl/%: $(BUILD_DESTDIR)/bcl/monotouch_tv_runtime/% | $(PREFIX)/lib/mono/Xamarin.TVOS/repl
+$(PREFIX)/lib/mono/Xamarin.TVOS/repl/%: $(BUILD_DESTDIR)/ios-bcl/monotouch_tv_runtime/% | $(PREFIX)/lib/mono/Xamarin.TVOS/repl
 	$(Q) install -m 0755 $< $@
 
-$(PREFIX)/lib/mono/Xamarin.TVOS/%: $(BUILD_DESTDIR)/bcl/monotouch_tv/% | $(PREFIX)/lib/mono/Xamarin.TVOS $(PREFIX)/lib/mono/Xamarin.TVOS/Facades
+$(PREFIX)/lib/mono/Xamarin.TVOS/%: $(BUILD_DESTDIR)/ios-bcl/monotouch_tv/% | $(PREFIX)/lib/mono/Xamarin.TVOS $(PREFIX)/lib/mono/Xamarin.TVOS/Facades
 	$(Q) install -m 0755 $< $@
 
 # copy to temporary directory before signing
 # otherwise we end up re-signing if installing into a different directory.
 
 TMP_BCL_TARGET_DIRS = \
-	$(BUILD_DESTDIR)/bcl/monotouch \
-	$(BUILD_DESTDIR)/bcl/monotouch/Facades \
-	$(BUILD_DESTDIR)/bcl/monotouch_runtime \
-	$(BUILD_DESTDIR)/bcl/monotouch_tv \
-	$(BUILD_DESTDIR)/bcl/monotouch_tv/Facades \
-	$(BUILD_DESTDIR)/bcl/monotouch_tv_runtime \
-	$(BUILD_DESTDIR)/bcl/monotouch_watch \
-	$(BUILD_DESTDIR)/bcl/monotouch_watch/Facades \
-	$(BUILD_DESTDIR)/bcl/monotouch_watch_runtime \
-	$(BUILD_DESTDIR)/bcl/xammac \
-	$(BUILD_DESTDIR)/bcl/xammac/Facades \
-	$(BUILD_DESTDIR)/bcl/xammac_net_4_5 \
-	$(BUILD_DESTDIR)/bcl/xammac_net_4_5/Facades \
-
-$(BUILD_DESTDIR)/bcl/%.dll: $(MONO_PATH)/mcs/class/lib/%.dll | $(TMP_BCL_TARGET_DIRS)
+	$(BUILD_DESTDIR)/ios-bcl/monotouch \
+	$(BUILD_DESTDIR)/ios-bcl/monotouch/Facades \
+	$(BUILD_DESTDIR)/ios-bcl/monotouch_runtime \
+	$(BUILD_DESTDIR)/ios-bcl/monotouch_tv \
+	$(BUILD_DESTDIR)/ios-bcl/monotouch_tv/Facades \
+	$(BUILD_DESTDIR)/ios-bcl/monotouch_tv_runtime \
+	$(BUILD_DESTDIR)/ios-bcl/monotouch_watch \
+	$(BUILD_DESTDIR)/ios-bcl/monotouch_watch/Facades \
+	$(BUILD_DESTDIR)/ios-bcl/monotouch_watch_runtime \
+	$(BUILD_DESTDIR)/mac-bcl/xammac \
+	$(BUILD_DESTDIR)/mac-bcl/xammac/Facades \
+	$(BUILD_DESTDIR)/mac-bcl/xammac_net_4_5 \
+	$(BUILD_DESTDIR)/mac-bcl/xammac_net_4_5/Facades \
+
+$(BUILD_DESTDIR)/ios-bcl/%.dll: $(MONO_IOS_SDK_DESTDIR)/ios-bcl/%.dll | $(TMP_BCL_TARGET_DIRS)
+	$(Q) $(CP) $< $@
+	$(Q_SN) MONO_CFG_DIR=$(TOP) $(SYSTEM_SN) -q -R $@ $(PRODUCT_KEY_PATH)
```

Comment:
do we need to re-sign here? the assemblies from the mono archive should already have the correct signature (if they don't I'd consider that a bug).

Reply:
I don't think we do need to re-sign. I'll have a look and put up another PR if that's the case.